src/app/actions/priceChart.js

```diff
@@ -0,0 +1,26 @@
+import { newScopedCreateAction } from 'Utilities/action'
+import log from 'Utilities/log'
+import Faast from 'Services/Faast'
+
+import { isPriceChartLoading, isPriceChartStale, getPriceChartData } from 'Selectors'
+
+const createAction = newScopedCreateAction(__filename)
+
+export const priceChartDataLoading = createAction('PRICE_CHART_DATA_LOADING', (symbol) => (symbol))
+export const priceChartDataUpdated = createAction('PRICE_CHART_DATA_UPDATED', (symbol, priceData) => ({ symbol, priceData }))
+export const priceChartDataError = createAction('PRICE_CHART_DATA_ERROR', (symbol, priceError) => ({ symbol, priceError }))
+
+export const fetchPriceChartData = (symbol) => (dispatch, getState) => {
+  if (isPriceChartLoading(getState(), symbol) || !isPriceChartStale(getState(), symbol)) {
+    return getPriceChartData(getState(), symbol)
```

Comment:
This should return the data in a promise like below

Reply:
doesn't it not return the data it just updates the reducer which is the received by the connect() in the component?

Reply:
So technically your thunks can be called like this:
`let x = dispatch(fetchPriceChartData)`
and x will be whatever you return from the thunk, so if you returned a promise with the data you could use it like so:
`dispatch(fetchPriceChartData).then((priceChartData) => do something with the data)`
I don't think it's best practice to do that though so I generally avoid returning anything useful from my actions except for under some exceptional circumstances.